cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -425,7 +425,7 @@ private ModelCompletenessRequirements modelCompletenessRequirements(Collection<G
   /**
    * Get a goals by priority based on the goal list.
    */
-  private Map<Integer, Goal> goalsByPriority(List<String> goals) {
+  public Map<Integer, Goal> goalsByPriority(List<String> goals) {
```

Comment:
Can we expose an interface to check if goals are ready instead of exposing the `goalsByPriority()` method here? Also, even if the requirement is not met for a goal, we may still want to be able to fix the cluster using `withAvailableValidWindows`.